src/main/drivers/accgyro_mpu.c

```diff
@@ -234,8 +234,7 @@ static bool detectSPISensorsAndUpdateDetectionResult(gyroDev_t *gyro, const sens
 #else
     UNUSED(bus);
 #endif
-    if (mpu6000SpiDetect(&gyro->bus)) {
-        gyro->mpuDetectionResult.sensor = MPU_60x0_SPI;
+    if ((gyro->mpuDetectionResult.sensor = mpu6000SpiDetect(&gyro->bus))) {
```

Comment:
Did you check this out?
https://github.com/martinbudden/cleanflight/blob/bf_gyro_spi_selection/src/main/sensors/gyro.c#L214-L226
```
            switch(dev->mpuDetectionResult.sensor) {
            case MPU_9250_SPI:
                gyroHardware = GYRO_MPU9250;
                break;
            case ICM_20608_SPI:
                gyroHardware = GYRO_ICM20608G;
                break;
            case ICM_20602_SPI:
                gyroHardware = GYRO_ICM20602;
                break;
            default:
                gyroHardware = GYRO_MPU6500;
            }
```